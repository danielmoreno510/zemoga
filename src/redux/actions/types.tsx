// Rulings --------------------------------------------------------------------------
export enum RULINGS {
    FETCH_RULING_LIST = 'FETCH_RULING_LIST',
    PUT_RULING_LIST = 'PUT_RULING_LIST',
    PUT_OPTION_RULING_LIST = 'PUT_OPTION_RULING_LIST',
    FETCH_RULING_DETAILS = 'FETCH_RULING_DETAILS',
    PUT_RULING_DETAILS = 'PUT_RULING_DETAILS',
    FETCH_RULING_OPERATION = 'FETCH_RULING_OPERATION',
    PUT_RULING_OPERATION= 'PUT_RULING_OPERATION'
}