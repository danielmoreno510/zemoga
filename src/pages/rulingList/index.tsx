import RulingListContainer from './rulingListContainer';

export default RulingListContainer;
