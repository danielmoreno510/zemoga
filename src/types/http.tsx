interface IHttpError {
  error: {
    message: string;
    name: string;
    stack: string;
    config: {
      transitional: {
        silentJSONParsing: true;
        forcedJSONParsing: true;
        clarifyTimeoutError: false;
      };
      adapter: string[];
      transformRequest: any[];
      transformResponse: any[];
      timeout: number;
      xsrfCookieName: string;
      xsrfHeaderName: string;
      maxContentLength: number;
      maxBodyLength: number;
      env: any;
      headers: {
        Accept: string;
      };
      baseURL: string;
      withCredentials: boolean;
      method: string;
      url: string;
    };
    code: string;
    status: number;
  };
}
