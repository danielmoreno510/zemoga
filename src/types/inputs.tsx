export enum OPTIONS {
    GRID = 'Grid',
    LIST = 'List',
  }

  export enum OPERATION {
    ADD = 'ADD',
    SUBS = 'SUBS',
    NULL = ''
  }