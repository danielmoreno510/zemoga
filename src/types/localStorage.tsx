import { IRuling } from "./rulings";

export enum KeysEnum {
  RulingList = 'RulingList',
  RulingDetails = 'RulingDetails',
}

export interface IStorageRulingDetails {
  expiry: number;
  data: IRuling;
}
