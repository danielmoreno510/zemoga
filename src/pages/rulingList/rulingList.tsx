import InputSearch from "@/components/InputSelect";
import RulingCard from "@/components/RulingCard";
import LoadingIndicator from "@/components/common/LoadingIndicator";
import { OPERATION, OPTIONS } from "@/types/inputs";
import {
  IRulingListDispatchProps,
  IRulingInitialState,
  IRuling,
} from "@/types/rulings";
import React from "react";

export const RulingList: React.FC<IRulingInitialState & IRulingListDispatchProps> = ({
  rulingList,
  option,
  changeOption,
  fetchVote,
}) => {
  const optionRuling = (value: OPTIONS) => changeOption!(value);
  const calculateVote = (operation: OPERATION, ruling: IRuling) => {
    const rulingUpdated: IRuling = rulingList!.find((currentRuling) => currentRuling.id === ruling.id)!;
    fetchVote!(operation, rulingUpdated);
  };

  return (
    <div>
      <div className="flex justify-end items-baseline">
        <InputSearch data-testid="inputSearch"  value={option!} setValue={optionRuling} />
      </div>
      <div className="card-list">
        {(rulingList || []).map((ruling) => (
          <RulingCard
            data-testid="rulingCard"
            ruling={ruling}
            key={ruling.name}
            view={option!}
            saveVote={(operation) => calculateVote!(operation, ruling)}
          />
        ))}
      </div>
    </div>
  );
};

const ListWithLoading = LoadingIndicator(RulingList);

const RulingListWithLoading: React.FC<
  IRulingInitialState & IRulingListDispatchProps
> = ({
  isFetchingRulingList,
  rulingList,
  option,
  getRulings,
  changeOption,
  fetchVote,
}) => {
  React.useEffect(() => {
    getRulings!();
  }, [getRulings]);

  return (
    <ListWithLoading
      isLoading={isFetchingRulingList!}
      option={option}
      rulingList={rulingList}
      changeOption={changeOption}
      fetchVote={fetchVote}
    />
  );
};

export default RulingListWithLoading;
