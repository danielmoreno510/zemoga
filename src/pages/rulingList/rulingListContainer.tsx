import RulingList from './rulingList';
import { StoreService } from '@/redux/StoreService';
import { fetchRulingList, putOptionRulingList, fetchRulingOperation } from '@/redux/actions';
import { IRulingReducer, IRulingListDispatchProps, IRulingListStateProps, IRulingInitialState, IRuling } from '@/types/rulings';
import { getRulingList } from '../../redux/selectors/rulingsSelectors';
import { OPERATION, OPTIONS } from '@/types/inputs';

export const mapStateToProps = (state: IRulingListStateProps): IRulingInitialState => ({
  isFetchingRulingList: state.rulings.isFetchingRulingList,
  rulingList: getRulingList(state),
  option: state.rulings.option,
});

export const mapDispatchToProps = (dispatch: (T: IRulingReducer) => void): IRulingListDispatchProps => ({
  getRulings: () => {
    dispatch(fetchRulingList());
  },
  changeOption: (option: OPTIONS) => {
    dispatch(putOptionRulingList(option));
  },
  fetchVote: (operation: OPERATION, ruling: IRuling) => {
    dispatch(fetchRulingOperation(operation, ruling));
  },
});

const RulingListContainer = StoreService.connect(mapStateToProps, mapDispatchToProps)(RulingList);

export default RulingListContainer;
