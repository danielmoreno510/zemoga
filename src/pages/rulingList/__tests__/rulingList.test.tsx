import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { RulingList } from "../rulingList";
import { OPTIONS } from "@/types/inputs";
import db from "../../../../db.json"

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))

describe("rulingList component", () => {
  describe("GIVEN a set of valid props", () => {
    const testProps = {
      rulingList: db.data,
      option: OPTIONS.GRID,
      changeOption: jest.fn(),
      fetchVote: jest.fn(),
    };

    describe("WHEN the component is rendered", () => {
      const container = render(<RulingList {...testProps} />);
      const open = container.getAllByTestId("open");
      fireEvent.click(open[0]);

      it("THEN should display a regular Home", () => {
        expect(container.container).toMatchSnapshot();
      });
    });
  });
});
