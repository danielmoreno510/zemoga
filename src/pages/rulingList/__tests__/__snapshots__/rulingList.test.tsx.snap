// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rulingList component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Home 1`] = `
<div>
  <div>
    <div
      class="flex justify-end items-baseline"
    >
      <div>
        <button
          class="text-black border-2 border-black border-solid outline-none font-medium text-sm px-5 py-2.5 text-center inline-flex items-center"
          data-dropdown-delay="500"
          data-dropdown-toggle="dropdownDelay"
          data-dropdown-trigger="hover"
          data-testid="open"
          id="dropdownDelayButton"
          type="button"
        >
          Grid
           
          <svg
            aria-hidden="true"
            class="w-2.5 h-2.5 ml-2.5"
            fill="none"
            viewBox="0 0 10 6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1 1 4 4 4-4"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="z-10 bg-white divide-y divide-gray-100 border-2 border-black border-solid border-t-white shadow"
          id="dropdownDelay"
        >
          <ul
            aria-labelledby="dropdownDelayButton"
            class="py-2 text-sm text-gray-700"
          >
            <li
              class="block px-4 py-2 text-black cursor-pointer"
              data-testid="list"
            >
              List
            </li>
            <li
              class="block px-4 py-2 text-black cursor-pointer"
              data-testid="grid"
            >
              Grid
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="card-list"
    >
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Pope Francis?
            </div>
            <p
              class="ellipsis-grid"
            >
              Heâ€™s talking tough on clergy sexual abuse, or is he just another pervert protector? (thumbs down) or a true pedophile punishing pontiff? (thumbs up).
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Kanye West
            </div>
            <p
              class="ellipsis-grid"
            >
              Born in Atlanta and raised in Chicago, West was first known as a producer for Roc-A-Fella Records in the early 2000s, producing singles for several mainstream artists.
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Mark Zuckerberg
            </div>
            <p
              class="ellipsis-grid"
            >
              Born in White Plains, New York, Zuckerberg attended Harvard University, where he launched the Facebook social networking service from his dormitory room on February 4, 2004.
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Cristina FernÃ¡ndez de Kirchner
            </div>
            <p
              class="ellipsis-grid"
            >
              Her first term of office started with a conflict with the agricultural sector, and her proposed taxation system was rejected.
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Malala Yousafzai
            </div>
            <p
              class="ellipsis-grid"
            >
              The daughter of educational activist Ziauddin, Yousafzai was born to a Pashtun family in Mingora, Khyber Pakhtunkhwa, Pakistan. Her family came to run a chain of schools in the region.
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Elon Musk
            </div>
            <p
              class="ellipsis-grid"
            >
              In 2002, Musk founded SpaceX, an aerospace manufacturer and space transport services company, of which he is CEO, CTO, and lead designer.
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container-grid"
      >
        <img
          alt="test"
          class="image-grid"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          sizes="(min-width: 750px) 1440px, 100vw"
          src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
          style="color: transparent;"
          width="0"
        />
        <div
          class="section-grid z-20 w-full absolute h-28 items-center"
        >
          <div
            class="content-grid"
          >
            <div
              class="text-xl mb-2 cursor-pointer"
            >
              Greta Thumberg
            </div>
            <p
              class="ellipsis-grid"
            >
              Thunberg's activism started after convincing her parents to adopt several lifestyle choices to reduce their own carbon footprint.
            </p>
          </div>
          <div
            class="vote-container-grid pr-6"
          >
            <div
              class="text-end text-white mb-2"
            >
              1 month ago
            </div>
            <div
              class="flex justify-end"
            >
              <button
                aria-label="thumbs up"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs up"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-up.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs down"
                class="icon-button !p-2 mr-4 "
              >
                <img
                  alt="thumbs down"
                  class="vote-icon-grid"
                  data-nimg="1"
                  decoding="async"
                  height="0"
                  loading="lazy"
                  src="/thumbs-down.svg"
                  style="color: transparent;"
                  width="0"
                />
              </button>
              <button
                aria-label="thumbs up"
                class="vote-now-grid opacity-60 !cursor-auto"
                disabled=""
              >
                Vote Now
              </button>
            </div>
          </div>
        </div>
        <div
          class="opacity-grid bg-gradient-to-r from-neutral-600 from-1% via-neutral-800 via-50% to-neutral-600 to-99% relative z-10 opacity-60"
        />
        <div
          class="card-buttons-grid"
        >
          <div
            class="!ml-0 !mr-0 featured-card__buttons"
          >
            <div
              aria-label="thumbs up"
              class="icon-button items-center !opacity-100"
            >
              <img
                alt="thumbs up"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
            <div
              aria-label="thumbs down"
              class="icon-button items-center"
            >
              <img
                alt="thumbs down"
                class="card-image-grid"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
              <div
                class="ml-1"
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
